components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/util/Util.java

```diff
@@ -286,4 +294,59 @@ public static RealmService getRealmService() {
         return Util.realmService;
     }
 
+    /** Build NameIDPolicy object given name ID policy format
+     * 
+     * @param nameIdPolicy Name ID policy format
+     * @return
+     */
+    public static NameIDPolicy buildNameIDPolicy(String nameIdPolicy) {
+        NameIDPolicy nameIDPolicyObj = new NameIDPolicyBuilder().buildObject();
+        if (!StringUtils.isEmpty(nameIdPolicy)){
+            nameIDPolicyObj.setFormat(nameIdPolicy);
+        }else {
+            nameIDPolicyObj.setFormat(SSOConstants.NAME_ID_POLICY_DEFAULT);
+        }
+        nameIDPolicyObj.setAllowCreate(true);
+        return nameIDPolicyObj;
+    }
+
+    /** Build NameID object given name ID format
+     * 
+     * @param nameIdFormat Name ID format
+     * @param subject
+     * @return
+     */
+    public static NameID buildNameID(String nameIdFormat, String subject) {
+        NameID nameIdObj = new NameIDBuilder().buildObject();
+        if (!StringUtils.isEmpty(nameIdFormat)) {
+            nameIdObj.setFormat(nameIdFormat);
+        } else {
+            nameIdObj.setFormat(SSOConstants.NAME_ID_POLICY_DEFAULT);
+        }
+        nameIdObj.setValue(subject);
+        return nameIdObj;
+    }
+
+    /**
+     * Replaces the ${} in url with system properties and returns
+     *
+     * @param acsUrl
+     * @return
+     */
+    public static String processAcsUrl(String acsUrl){
```

Comment:
Fixed in #1581 (https://github.com/wso2/carbon-apimgt/commit/0e5a93cca133738cc74a26863424e20db485892d)